user_agreement_core_lib/data_layers/data_access/user_agreement_document_data_access.py

```diff
@@ -17,21 +17,45 @@ class UserAgreementDocumentDataAccess(DataAccess):
     def __init__(self, db: SqlAlchemyConnectionRegistry):
         self.db_session = db
 
-    def agree(self, user_id: int, document_id: int):
+    def set_agreement(self, user_id: int, document_id: int, is_agreed: bool = False):
         with self.db_session.get() as session:
-            entity = UserAgreementDocument()
-            entity.user_id = user_id
-            entity.agreement_document_id = document_id
-            session.add(entity)
-        return entity
+            existing = (
```

Comment:
move to service 
if agees and already agrred do nothing 
if agree or dissagree diffrent from current state. 

delete last  + crate new